packages/core/src/main.ts

```diff
@@ -149,14 +150,11 @@ export async function createLibp2pInstance(
   libp2p._dht._lan._topologyListener._protocol = HOPR_DHT_LAN_PROTOCOL
 
   const onConnectionOriginal = libp2p.upgrader.onConnection
-  const onConnection = async (conn: Connection) => {
-    // ensure peer is allowed to connect
-    if (await isAllowedToConnect(conn.remotePeer)) {
-      onConnectionOriginal(conn)
-    }
-  }
-  libp2p.upgrader.onConnection = (conn: Connection) => {
-    onConnection(conn)
+  libp2p.upgrader.onConnection = async (conn: Connection) => {
+    // check if we are denying connection to that peer
+    if (isDenied && (await isDenied(conn.remotePeer))) return
```

Comment:
```suggestion
    if (isDenied && (await isDenied(conn.remotePeer))) {
      try {
        await conn.close()
      } catch (err: any) {
        log(`Error while closing connection to non-whitelisted node`, err)
      }
      return
    }
```
Forgot to mention that we need to *close* these connections as they otherwise live forever

Reply:
thanks